### DISCUSSION TITLE
Scripting Geany with JavaScript?

### DISCUSSION THREAD

**Discussion by ghost** [geany/geany#3972] [General]
Geany Plugins can be written in Python and Lua as I know. What about JavaScript? There is a project named GNOME JavaScript that provide the ability to write GTK applications in pure JavaScript. Can it be used with Geany?

**Comment by elextr** [comment]
There is currently no way of writing Geany plugins in Javascript.

**Comment by b4n** [comment]
Not sure what the state of [Peasy](https://github.com/kugel-/peasy) is, but it likely could work -- it only advertises Python and Lua, but I though libpeas supported JavaScript as well… not sure.

**Reply by elextr** [reply]
I checked libpeas before my comment, it only supports " (currently, C, Lua 5.1,
and Python 3)", no Javascript.

**Reply by b4n** [reply]
The documentation might be outdated, see https://gitlab.gnome.org/GNOME/libpeas/-/tree/main/loaders?ref_type=heads

**Reply by elextr** [reply]
Well if it isn't documented it does not exist!!!!!!! :grin:

Edit: also Peasy is libpeas 1 IIUC, and something added undocumented to libpeas 2 may not be supported until its updated.
